packages/cactus-plugin-satp-hermes/src/main/typescript/gol/satp-manager.ts

```diff
@@ -297,10 +304,157 @@ export class SATPManager {
       if (index < handlersClasses.length) {
         const HandlerClass = handlersClasses[index];
         const handler = new HandlerClass(options);
-        handlers.push(handler);
+        this.satpHandlers.set(handler.getHandlerIdentifier(), handler);
       }
     });
+  }
+
+  public async initiateTransfer(session: SATPSession): Promise<void> {
+    const fnTag = `${SATPManager.CLASS_NAME}#initializeHandlers()`;
+    this.logger.info(`${fnTag}, Initiating Transfer`);
+    this.logger.debug(
+      `SessionData: ${JSON.stringify(session.getClientSessionData())}`,
+    );
+
+    if (!session.getClientSessionData()) {
+      throw new Error(`${fnTag}, Session not found`);
+    }
+
+    const channel = this.orquestrator.getChannel(
+      session.getClientSessionData()
+        ?.recipientGatewayNetworkId as SupportedChain,
+    );
+
+    if (!channel) {
+      throw new Error(`${fnTag}, Channel not found`);
+    }
+
+    const sessionData: SessionData =
+      session.getClientSessionData() as SessionData;
+
+    const clientSatpStage0: PromiseConnectClient<typeof SatpStage0Service> =
+      channel.clients.get("0") as PromiseConnectClient<
+        typeof SatpStage0Service
```

Comment:
Seems to be a problem here with the type narrowing

Reply:
this variables are just for each stage wouldnt it be better if the type is only of that service?